_posts/2017-02-09-rapid-mysql-imports-docker.md

```diff
@@ -0,0 +1,186 @@
+
+## Bonus points: integrate this process with your continuous integration workflow
+
+The above steps can be manually performed by a technical lead on the project, who is responsible for generating and distributing the MySQL data volume to team members and your testing infrastructure.
+
+But we can get further productivity gains by automating this process completely with [Travis CI](https://www.travis-ci.com) and GitHub hooks. In outline, here's what this process looks like:
+
+### 1. Generate a new seed database SQL dump after production deployments
+
+We use [Fabric](http://www.fabfile.org) for deploying to staging and production. When we deploy to production (production is not on a Docker stack), our post-deployment tasks include generating a traditional MySQL database dump and copying this to Amazon S3:
+
+``` python
+def update_seed_db():
+    run('drush -r %s/www/web sql-dump \
+    --result-file=/tmp/$(date +%%Y-%%m-%%d)--post-deployment.sql --gzip \
+    --structure-tables-list=cache,cache_*,history,search_*,sessions,watchdog' \
+    % env.code_dir)
+    run('/usr/local/bin/aws s3 cp /tmp/$(date +%Y-%m-%d)--post-deployment.sql.gz \
+    s3://{bucket-name}/seed-database/database.sql.gz --sse')
+    run('rm /tmp/$(date +%Y-%m-%d)--post-deployment.sql.gz')
+```
+
+### 2. When work is merged into develop, create a new MySQL data volume archive
+
+We use [`git flow`](http://developers.savaslabs.com/en/latest/git.html#basic-git-workflow-git-flow) on just about all of our Drupal projects. Whenever a feature branch is merged into `develop`, we update the MySQL data volume archive dump for use in Travis CI tasks and local development. First, there is a specification in our `.travis.yml` file that calls a deployment script:
+
+``` yaml
+deploy:
+  provider: script
+  script:
+    - resources/scripts/travis-deploy.sh
+  skip_cleanup: true
+  on:
+    branch: develop
+```
+
+And the `travis-deploy.sh` script:
+
+``` bash
+#!/usr/bin/env bash
+
+set -e
+
+make import-seed-db
+make export-mysql-data
+aws s3 cp db-data-volume.tar.gz \
+s3://{bucket-name}/mysql-data-volume/db-data-volume.tar.gz --sse
+```
+
+This script: (1) imports the traditional MySQL seed database file from production, and then (2) creates a MySQL data volume archive. (We use a [`Makefile`](https://github.com/savaslabs/docker-drupal-template/blob/master/Makefile) for simplifying some of the common site provisioning tasks for developers and our CI systems; more on that in another post!)
```

Comment:
Should we consider showing the `make import-seed-db` Makefile example to clarify things (as it contains the actual import step talked about in this blog?)? I can see the argument for not including it now to save it for the next post, on the other hand, it might tie things together nicely in here.